### PR TITLE
Sc 12805/query by id

### PR DESCRIPTION
**Pull request by yarden-slon**
[sc-12805]

### CODE REVIEW

**Comment by rajat08** [comment]
@yarden-slon  can you add a little detail about this in the changelog? You can create a changed header and put things under it

**Comment by rajat08** [comment]
> 

not necessary but good to have

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 ### Changed
 - Added support for deleting vectors by metadata filter. The pinecone.Index.delete() api now accepts an additional filter= parameter which takes metadata filter expression equivalent to what query() supports.
   - Internally these requests are now sent as POST requests, though the previous DELETE api is still supported.
+- Added support of querying by vector id.
```
Comment (rajat08) [inline CHANGELOG.md:7]: We have a release for 2.0.10 can you put it under a new Unreleased header?

File: tests/unit/test_index.py [hunk 2]
```diff
@@ -43,7 +43,7 @@ def test_query_request_tuples_query_only(mocker):
     mocker.patch.object(index._vector_api, 'query', autospec=True)
     index.query(top_k=10, vector=vals1)
     index._vector_api.query.assert_called_once_with(
-        pinecone.QueryRequest(top_k=10, vector=vals1, queries=[])
+        pinecone.QueryRequest(top_k=10, vector=vals1, id='', queries=[])
```
[the remark this answers is not in the document]
Comment (mutayroei) [reply tests/unit/test_index.py:46]: or - was it auto generated?

[the remark this answers is not in the document]
Comment (yarden-slon) [reply tests/unit/test_index.py:46]: Not auto-generated. But empty works fine. And I think if the rest of the args are not None I think it should be the same.

File: CHANGELOG.md (hunk 1 above)
Comment (yarden-slon) [reply CHANGELOG.md:7]: Didn't know that. Updateing

**Pull request merged**